doc/source/user_guide/index.rst

```diff
@@ -88,3 +88,4 @@ Guides
     sparse
     gotchas
     cookbook
+    user_defined_functions
```

Comment:
I would move this before the `groupby` section. It feels more natural to me to explain first `Series.apply` and later explain `groupby("col").apply`.